AmplifyPlugins/Core/AWSPluginsCore/Model/Support/SelectionSet.swift

```diff
@@ -8,104 +8,102 @@
 import Foundation
 import Amplify
 
-/// Contains detail about the current selection set. `pagination` indicates that it contains paginated fields.
-/// `modelField` indicates that the values represent the property names from a `ModelSchema`.
-public enum SelectionSetType {
-    case pagination
-    case modelField
-}
+public typealias SelectionSet = TreeNode<SelectionSetField>
 
-/// A nested structure to contain a list of `SelectionSetField`, each of which may optionally contain another
-/// `SelectionSet`
-public struct SelectionSet {
-    var fields: [SelectionSetField]
-    var type: SelectionSetType
+public class TreeNode<E> {
+    var value: E
+    var children: [TreeNode<E>] = []
+    weak var parent: TreeNode<E>?
 
-    public init(fields: [SelectionSetField], type: SelectionSetType) {
-        self.fields = fields
-        self.type = type
+    init(value: E) {
+        self.value = value
     }
 
-    mutating func append(_ selectionSetField: SelectionSetField) {
-        fields.append(selectionSetField)
+    func add(child: TreeNode) {
```

Comment:
This has the non-obvious side-effect of mutating the child. Suggest you rename the method to make it more obvious to `addChild(settingParentOf child:)` or similar, and add a doc comment with a note. It's the right behavior--it's would be cumbersome to require call sites to do this explicitly--but we just want to make sure that it's obvious what's happening when we call the method.

Reply:
thanks, updating!